libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -822,6 +825,23 @@ string YulUtilFunctions::arrayAllocationSizeFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::structAllocationSizeFunction(StructType const& _type)
+{
+	solAssert(_type.dataStoredIn(DataLocation::Memory), "");
+	string functionName = "struct_allocation_size_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		Whiskers w(R"(
+			function <functionName>(memoryMemberCount) -> size {
```

Comment:
If the type is part of the function name, why is the member count a parameter?

This function should return `_type.memoryDataSize()` shouldn't it?

Reply:
Yes for both